src/serialize/DynamicObject.js

```diff
@@ -0,0 +1,170 @@
+"use strict";
+
+
+const Point= require('../Point');
+const Serializable= require('./Serializable');
+const Serializer= require('./Serializer');
+
+/**
+ * Defines an objects which can move about in the game world
+ */
+class DynamicObject extends Serializable {
+
+    static get netScheme(){
+        return {
+            id: { type: Serializer.TYPES.UINT8 },
+            playerId: { type: Serializer.TYPES.UINT8 },
+            x: { type: Serializer.TYPES.INT16 },
+            y: { type: Serializer.TYPES.INT16 },
+            velX: { type: Serializer.TYPES.FLOAT32 },
+            velY: { type: Serializer.TYPES.FLOAT32 },
+            angle: { type: Serializer.TYPES.INT16 }
+        }
+    }
+
+    constructor(id, x, y){
+        super();
+        this.id = id; //instance id
+        this.playerId = 0;
+        this.x = x;
+        this.y = y;
+        this.velX = 0;
+        this.velY = 0;
+        this.angle = 90;
+        this.rotationSpeed = 3;
+        this.acceleration = 0.1;
+        this.deceleration = 0.99;
+        this.maxSpeed = 2;
+
+        //todo deal with what goes over the wire
+        this.velocity = new Point();
+        this.temp={
+            accelerationVector: new Point()
+        };
+
+    };
+
+    copyFrom(sourceObj){
+        this.id = sourceObj.id;
+        this.playerId = sourceObj.playerId;
+        this.isPlayerControlled = sourceObj.isPlayerControlled;
+
+        this.x = sourceObj.x;
+        this.y = sourceObj.y;
+        this.velX = sourceObj.velX;
+        this.velY = sourceObj.velY;
+        this.velocity.set(sourceObj.velX, sourceObj.velY);
+        this.angle = sourceObj.angle;
+        this.rotationSpeed = sourceObj.rotationSpeed;
+        this.acceleration = sourceObj.acceleration;
+        this.deceleration = sourceObj.deceleration;
+        this.maxSpeed = sourceObj.maxSpeed;
+    }
+
+    step(worldSettings){
+        if (this.isRotatingRight){ this.angle += this.rotationSpeed; }
+        if (this.isRotatingLeft){this.angle -= this.rotationSpeed; }
+
+        if(this.angle>360){ this.angle -= 360; }
+        if(this.angle<0){ this.angle += 360; }
+
+        if (this.isAccelerating) {
```

Comment:
Is it all spaaace-specific? Can we carve out some part which is generic for an object in two dimensional space, with an orientation, acceleration and deceleration constants, etc. ?